integration-tests/lts/select.test.ts

```diff
@@ -1136,6 +1136,35 @@ SELECT __scope_0_defaultPerson {
     );
   });
 
+  test.skip("polymorphic from type intersection", async () => {
+    const query = e.select(e.Movie.characters, (person) => ({
+      heroMovieCount: person.is(e.Hero).number_of_movies,
+      heroInfo: e.select(person.is(e.Hero), (hero) => ({
+        number_of_movies: true,
+        numMovies: hero.number_of_movies,
+      })),
+    }));
```

Comment:
This PR doesn't yet create a discriminated union from this 😢

Reply:
This is result is the same in master. Since it hasn't regressed in this PR, I'll leave it for a separate PR.